lighthouse-core/gather/computed/dependency-graph/estimator/estimator.js

```diff
@@ -0,0 +1,284 @@
+      TcpConnection.maximumSaturatedConnections(this._rtt, this._throughput),
+      this._options.maximumConcurrentRequests
+    );
+  }
+
+  /**
+   * @param {!WebInspector.NetworkRequest} record
+   * @return {number}
+   */
+  static getResponseTime(record) {
+    const timing = record._timing;
+    return (timing && timing.receiveHeadersEnd - timing.sendEnd) || Infinity;
+  }
+
+  _initializeNetworkRecords() {
+    const records = [];
+
+    this._graph.getRootNode().traverse(node => {
+      if (node.type === Node.TYPES.NETWORK) {
+        records.push(node.record);
+      }
+    });
+
+    this._networkRecords = records;
+    return records;
+  }
+
+  _initializeNetworkConnections() {
+    const connections = new Map();
+    const recordsByConnection = groupBy(
+      this._networkRecords,
+      record => record.connectionId
+    );
+
+    for (const [connectionId, records] of recordsByConnection.entries()) {
+      const isSsl = records[0].parsedURL.scheme === 'https';
+      let responseTime = records.reduce(
+        (min, record) => Math.min(min, Estimator.getResponseTime(record)),
+        Infinity
+      );
+
+      if (!Number.isFinite(responseTime)) {
+        responseTime = this._defaultResponseTime;
+      }
+
+      const connection = new TcpConnection(
+        this._rtt,
+        this._throughput,
+        responseTime,
+        isSsl
+      );
+
+      connections.set(connectionId, connection);
+    }
+
+    this._connections = connections;
+    return connections;
+  }
+
+  _initializeAuxiliaryData() {
+    this._nodeAuxiliaryData = new Map();
+    this._nodesCompleted = new Set();
+    this._nodesInProcess = new Set();
+    this._nodesInQueue = new Set(); // TODO: replace this with priority queue
+    this._connectionsInUse = new Set();
+  }
+
+  /**
+   * @param {!Node} node
+   */
+  _enqueueNodeIfPossible(node) {
+    const dependencies = node.getDependencies();
+    if (
+      !this._nodesCompleted.has(node) &&
+      dependencies.every(dependency => this._nodesCompleted.has(dependency))
+    ) {
+      this._nodesInQueue.add(node);
+    }
+  }
+
+  /**
+   * @param {!Node} node
+   * @param {number} totalElapsedTime
+   */
+  _startNodeIfPossible(node, totalElapsedTime) {
+    if (node.type !== Node.TYPES.NETWORK) return;
+
+    const connection = this._connections.get(node.record.connectionId);
+
+    if (
+      this._nodesInProcess.size >= this._maximumConcurrentRequests ||
+      this._connectionsInUse.has(connection)
+    ) {
+      return;
+    }
+
+    this._nodesInQueue.delete(node);
+    this._nodesInProcess.add(node);
+    this._nodeAuxiliaryData.set(node, {
+      startTime: totalElapsedTime,
+      timeElapsed: 0,
+      timeElapsedOvershoot: 0,
+      bytesDownloaded: 0,
+    });
+
+    this._connectionsInUse.add(connection);
+  }
+
+  _updateNetworkCapacity() {
+    for (const connection of this._connectionsInUse) {
+      connection.setThroughput(this._throughput / this._nodesInProcess.size);
```

Comment:
setThroughput vs availableThroughput makes me think there may be multiple... like currentTHroughput, defaultThroughput, etc... since there's only one should we just use that same name?

Reply:
yep done